deploy/01_resolve_nucypher_token.ts

```diff
@@ -0,0 +1,44 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+import { from1e18, to1e18 } from "../test/helpers/contract-test-helpers"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { getNamedAccounts, deployments, helpers } = hre
+  const { log } = deployments
+  const { deployer } = await getNamedAccounts()
+  const { execute, read } = deployments
+
+  const NuCypherToken = await deployments.getOrNull("NuCypherToken")
+
+  if (
+    hre.network.name === "mainnet" &&
+    NuCypherToken &&
+    helpers.address.isValid(NuCypherToken.address)
+  ) {
+    log(`using external NuCypherToken at ${NuCypherToken.address}`)
+    await deployments.save("NuCypherToken", NuCypherToken)
+  } else {
+    log(`deploying NuCypherToken stub`)
+
+    const NU_SUPPLY = to1e18("1000000000") // 1B NU
```

Comment:
This number should be changed if we want to make a testnet deployment that is closer to the real setting. We deployed the token with a 3.8 B supply, but most of it is on escrow in the `StakingEscrow` contract. The actual issued supply to date is around 1.34 B, and it will still be around that number when the merge happens.

Reply:
Good point! Please see 8096a55